src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -1759,10 +1759,17 @@ void XrdHttpProtocol::Cleanup() {
 
   if (ssl) {
 
-
-    if (SSL_shutdown(ssl) != 1) {
-      TRACE(ALL, " SSL_shutdown failed");
-      ERR_print_errors(sslbio_err);
+    int ret = SSL_shutdown(ssl);
```

Comment:
Don't you maybe need also a call to [`SSL_set_shutdown`](https://www.openssl.org/docs/man1.1.1/man3/SSL_set_shutdown.html) before like seen [here](https://github.com/xrootd/xrootd/blob/master/src/XrdTls/XrdTlsSocket.cc#L750)?

Reply:
Shouldn't SSL_shutdown anyway set this flag without need for us to set this. The docs do mention that 
"Whether the operation succeeds or not, the SSL_SENT_SHUTDOWN flag is set and a currently open session is considered closed and good and will be kept in the session cache for further reuse."